File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        Map<Integer, Integer> indexMap = new HashMap<>();

        int max = 0;
        for (int num : nums) {
            indexMap.put(num, indexMap.getOrDefault(num, 0) + 1);
            max = Math.max(max, indexMap.get(num));
        }
        for (int i = 0; i < max; i++) {
            result.add(new ArrayList<>());
        }

        for (Integer key : indexMap.keySet()) {
            int value = indexMap.get(key);
            while (value > 0) {
                result.get(value - 1).add(key);
                value--;
            }
        }

        return result;
    }
}

File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        result = []
        index_map = {}

        max_freq = 0
        for num in nums:
            index_map[num] = index_map.get(num, 0) + 1
            max_freq = max(max_freq, index_map[num])

        # Initialize the result list with empty lists
        for i in range(max_freq):
            result.append([])

        # Iterate over the keys of the index_map
        for key in index_map.keys():
            value = index_map[key]

            # Add the key to the corresponding rows in the result list
            while value > 0:
                result[value - 1].append(key)
                value -= 1

        return result 
        